From 2b70423942096380486711174d0d714e03983254 Mon Sep 17 00:00:00 2001
From: Andy Doan <andy@foundries.io>
Date: Wed, 10 Jul 2019 18:17:26 -0500
Subject: [PATCH 03/12] [FIO toup] aktualizr-lite: Use current target for
 status

Now that we are tracking the installed target properly, we can use
getCurrent to make this easier and more informative.

Signed-off-by: Andy Doan <andy@foundries.io>
---
 src/aktualizr_lite/main.cc      | 56 +++++++++++++++++++--------------
 src/aktualizr_lite/test_lite.sh |  7 +++++
 2 files changed, 39 insertions(+), 24 deletions(-)

diff --git a/src/aktualizr_lite/main.cc b/src/aktualizr_lite/main.cc
index 0243a62b..46bb2402 100644
--- a/src/aktualizr_lite/main.cc
+++ b/src/aktualizr_lite/main.cc
@@ -65,14 +65,41 @@ static std::shared_ptr<SotaUptaneClient> liteClient(Config &config, std::shared_
   return client;
 }
 
+static void log_info_target(const std::string &prefix, const Config &config, const Uptane::Target &t) {
+  auto name = t.filename();
+  if (t.custom_version().length() > 0) {
+    name = t.custom_version();
+  }
+  LOG_INFO << prefix + name << "\tsha256:" << t.sha256Hash();
+  if (config.pacman.type == PackageManager::kOstreeDockerApp) {
+    bool shown = false;
+    auto apps = t.custom_data()["docker_apps"];
+    for (Json::ValueIterator i = apps.begin(); i != apps.end(); ++i) {
+      if (!shown) {
+        shown = true;
+        LOG_INFO << "\tDocker Apps:";
+      }
+      if ((*i).isObject() && (*i).isMember("filename")) {
+        LOG_INFO << "\t\t" << i.key().asString() << " -> " << (*i)["filename"].asString();
+      } else {
+        LOG_ERROR << "\t\tInvalid custom data for docker-app: " << i.key().asString();
+      }
+    }
+  }
+}
+
 static int status_main(Config &config, const bpo::variables_map &unused) {
   (void)unused;
-  GObjectUniquePtr<OstreeSysroot> sysroot_smart = OstreeManager::LoadSysroot(config.pacman.sysroot);
-  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment(sysroot_smart.get());
-  if (deployment == nullptr) {
+  auto target = liteClient(config, nullptr)->getCurrent();
+
+  if (target == Uptane::Target::Unknown()) {
     LOG_INFO << "No active deployment found";
   } else {
-    LOG_INFO << "Active image is: " << ostree_deployment_get_csum(deployment);
+    auto name = target.filename();
+    if (target.custom_version().length() > 0) {
+      name = target.custom_version();
+    }
+    log_info_target("Active image is: ", config, target);
   }
   return 0;
 }
@@ -95,26 +122,7 @@ static int list_main(Config &config, const bpo::variables_map &unused) {
   for (auto &t : client->allTargets()) {
     for (auto const &it : t.hardwareIds()) {
       if (it == hwid) {
-        auto name = t.filename();
-        if (t.custom_version().length() > 0) {
-          name = t.custom_version();
-        }
-        LOG_INFO << name << "\tsha256:" << t.sha256Hash();
-        if (config.pacman.type == PackageManager::kOstreeDockerApp) {
-          bool shown = false;
-          auto apps = t.custom_data()["docker_apps"];
-          for (Json::ValueIterator i = apps.begin(); i != apps.end(); ++i) {
-            if (!shown) {
-              shown = true;
-              LOG_INFO << "\tDocker Apps:";
-            }
-            if ((*i).isObject() && (*i).isMember("filename")) {
-              LOG_INFO << "\t\t" << i.key().asString() << " -> " << (*i)["filename"].asString();
-            } else {
-              LOG_ERROR << "\t\tInvalid custom data for docker-app: " << i.key().asString();
-            }
-          }
-        }
+        log_info_target("", config, t);
         break;
       }
     }
diff --git a/src/aktualizr_lite/test_lite.sh b/src/aktualizr_lite/test_lite.sh
index 2a07522c..df57da54 100755
--- a/src/aktualizr_lite/test_lite.sh
+++ b/src/aktualizr_lite/test_lite.sh
@@ -106,3 +106,10 @@ OSTREE_HASH=$sha LD_PRELOAD=$mock_ostree $valgrind $aklite --loglevel 1 -c $sota
 ostree admin status
 
 OSTREE_HASH=$sha LD_PRELOAD=$mock_ostree $valgrind $aklite --loglevel 1 -c $sota_dir/sota.toml update | grep "Updating to: Target(zlast"
+
+out=$(OSTREE_HASH="$sha" LD_PRELOAD=$mock_ostree $valgrind $aklite --loglevel 1 -c $sota_dir/sota.toml status)
+if [[ ! "$out" =~ "Active image is: zlast	sha256:$sha" ]] ; then
+    echo "ERROR: status incorrect:"
+    echo $out
+    exit 1
+fi
-- 
2.22.0

